File: wrapped_kotlin_ported/funcs/f_6.kt
fun isSpam(message: String): Boolean {
    // Check for excessive use of special characters
    val specialChars = Regex("""[\*\.\(\)\\\-/@\[\]<>]""").findAll(message)
    if (specialChars.count() > 20) {
        return true
    }

    // Check for excessive use of numbers
    val numbers = Regex("""\d+""").findAll(message)
    if (numbers.count() > 15) {
        return true
    }

    // Check for pattern of shortened urls
    val urls = Regex("""(https?://[a-zA-Z0-9./]+)""").findAll(message)
    if (urls.count() > 5) {
        return true
    }

    // Check for presence of keywords in the message
    val keywords = arrayOf("상한가", "추천", "입장", "무료")
    keywords.forEach { keyword ->
        if (message.contains(keyword)) {
            return true
        }
    }
    return false
}

File: wrapped/funcs/f_6.py

import re

def is_spam(message):
    # Check for excessive use of special characters
    special_chars = re.findall(r'[\*\.\(\)\\\-/@\[\]<>]', message)
    if len(special_chars) > 20:
        return True

    # Check for excessive use of numbers
    numbers = re.findall(r'\d+', message)
    if len(numbers) > 15:
        return True

    # Check for pattern of shortened urls
    urls = re.findall(r'(https?://[a-zA-Z0-9./]+)', message)
    if len(urls) > 5: 
        return True

    # Check for presence of keywords in the message
    keywords = ['상한가', '추천', '입장', '무료']
    for keyword in keywords:
        if keyword in message:
            return True

    return False
